r1cs/gadgets/crypto/src/nizk/gm17/mod.rs

```diff
@@ -155,11 +155,13 @@ where
                 .by_ref()
                 .zip(pvk.query.iter().skip(1))
                 .enumerate()
-            {
-                let input_bits = input.to_bits(cs.ns(|| format!("Input {}", i)))?;
-                g_psi = b.mul_bits(cs.ns(|| format!("Mul {}", i)), &g_psi, input_bits.iter())?;
-                input_len += 1;
-            }
+                {
+                    let input_bits = input.to_bits(cs.ns(|| format!("Input {}", i)))?;
+                    g_psi = b
```

Comment:
?

Reply:
Interface of the mul_bits gadget has been changed from base.mul_bits(exponent, accumulator) to simply base.mul_bits(exponent). Accumulator now is initialized internally, but after we need to add g_psi, while before the accumulator was initialized directly to g_psi

Reply:
This often helps us dealing with exceptional cases internally, without the user needing to explicitly take care of it